doc/DEPLOYMENT_GUIDE.md

```diff
@@ -0,0 +1,398 @@
+# 背景
+
+本部署指南详细描述了如何在云上使用 AWS CloudFormation 模板部署 Keycloak，从而构建Keycloak on AWS 高可用架构。
+
+# 解决方案描述
+
+Keycloak 的容器在 [AWS Fargate](https://amazonaws-china.com/fargate/) 上部署和运行，借助于 AWS Fargate，您无需预置和管理容器实例。为了确保系统高可用，[Amazon ECS](https://amazonaws-china.com/ecs/) 服务定义了两个任务，如果某一个任务出现故障不能提供服务的情况下，另外的一个任务仍旧能对外提供服务。
+
+另外本解决方案使用 [Amazon RDS](https://amazonaws-china.com/rds/) 作为系统数据库，用于存储 Keycloak的配置及用户信息。为了确保系统的高可用性和信息安全。 默认采用了 [Amazon RDS 多可用区部署方式](https://aws.amazon.com/cn/rds/features/multi-az/) ，数据库在每个可用区中运行，并已设计为具备高可靠性。万一发生基础设施故障，Amazon RDS 可自动故障转移至备用实例中，以便您能够在故障转移结束后立即恢复数据库操作。此外，本解决方案也支持 [Amazon Aurora Serverless](https://aws.amazon.com/cn/rds/aurora/serverless/) 数据库集群的选项，Amazon Aurora Serverless 会根据应用程序的需求自动启动、关闭以及扩展或缩减容量，让您无需管理任何数据库实例，即可在云中运行数据库。
+
+无论选择哪一种数据库类型，缺省的数据库密码会在部署的同时自动产生并且存放在 [AWS Secrets Manager](https://aws.amazon.com/cn/secrets-manager/)。
+
+# 使用场景
+
+此解决方案提供了在AWS 云上构建高可用架构的Keycloak集群。Keycloak 是一款开箱即用的开源身份及访问控制软件，提供了单点登录（SSO）功能，支持OpenID Connect、OAuth 2.0、SAML 2.0标准协议。Keycloak 提供可自定义的用户界面，用于登录，注册，管理和帐户管理。此外，用户可以将Keycloak 与[Amazon Cognito](https://amazonaws-china.com/cognito/)或与其他现有的 LDAP 和 [Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/) 服务器进行集成。用户还可以将身份验证委派给第三方身份提供商。
+
+# 系统架构
+
+## 架构图
+此解决方案可在由西云数据运营的AWS（宁夏）区域或由光环新网运营的AWS（北京）区域中部署，也可部署在AWS其他海外Region。
+
+## 组件
+
+### Amazon ECS
+
+- 本解决方案使用 AWS Fargate 运行 Amazon ECS 集群，前者是容器的无服务器计算服务。使用 AWS Fargate，您无需预置和管理服务器，而且可以为每个应用程序指定资源并为其付费，并通过设计隔离应用程序来提高安全性。
+
+- 为了确保系统高可用，在Amazon ECS 服务定义了两个任务，如果某一个任务出现故障不能提供服务，另外的一个任务仍旧能继续提供服务。
+
+- Amazon ECS 支持 Docker 并使您能够运行和管理 Docker 容器。 Keycloak的容器镜像在 Amazon ECS 上部署和运行，无需进行任何配置更改。
+
+
+### Amazon ECR
+- 用于存储 Keycloak 的 Docker 镜像文件。
+Amazon Certificate Manager (ACM)
+- 用于 ALB 的 Https 侦听器（TLS：443）中 SSL Certificate。
+
+### Amazon Identity and Access Management (IAM)
+
+- Cognito Identity pool联合 Keycloak进行用户验证。其可以有两种 Role,一种是未登录用户所有拥有的 Role,一种是已经登录的用户所拥有的 Role。根据Role 所拥有的不同 Policy，用户可以访问不同的 AWS 服务。
+
+### Amazon Route 53
+- 用于域名解析及 ACM 证书创建时验证。
+Amazon RDS
+- 数据库类型是Amazon RDS for MySQL。
+- 默认数据库实例类型是 db.r5.large。
```

Comment:
@wchaws  can you confirm the default instance type?

Reply:
It's r5.large, I haven't add a parameter to control db instance yet